### PR TITLE
test(mcp): add WebSocket streaming integration tests

### PR DESCRIPTION
**Pull request by dhairyajangir**
# Add WebSocket Streaming Integration Tests for MCP Server

Source Branch: forge/9026-wish-test
issue - #225 

## Summary

Implements a comprehensive integration test suite for WebSocket streaming functionality in the Forge MCP server. The tests validate the full lifecycle of WebSocket connections including connection stability, data streaming with schema validation, error resilience, and proper cleanup for both client and server-initiated disconnects.

## Motivation

- **Gap in Test Coverage:** While the MCP server has OAuth2 integration tests, WebSocket streaming capabilities lacked automated validation
- **Real-Time Features:** WebSocket streaming is critical for real-time task monitoring, diff streaming, and log streaming
- **Reliability Assurance:** Ensures the `WebSocketManager` properly handles edge cases (errors, disconnects, cleanup)
- **Regression Prevention:** Prevents future breaks in WebSocket connection pooling, auto-reconnection, and subscription management

### New File: `forge/mcp/tests/integration/websocket-streaming.test.ts` (+353 lines)

**Test Suite: "MCP WebSocket Streaming Validation"**

#### Test 1: Connection Stability ✅
- Validates WebSocket connection establishment via `WebSocketManager`
- Verifies connection remains stable for minimum 75ms duration
- Confirms clean closure with valid status codes (1000 or 1005)
- Tests basic subscribe/unsubscribe flow

#### Test 2: Streaming Data Validation ✅
- Simulates server streaming 5 JSON-formatted message chunks
- Validates message schema conformance:
  - Required field: `type` (e.g., "stream-chunk")
  - Required field: `sessionId` (session identifier)
  - Required field: `data` (payload object with `index` and `payload` properties)
- Confirms all messages received in correct order
- Verifies data types (number for index, string for payload)

#### Test 3: Error Handling & Resilience ✅
- **Server-side errors:** Simulates server sending error frames, validates graceful handling
- **Invalid client payloads:** Tests manager's response to malformed JSON from client
- **Connection stability:** Verifies connection remains open during error conditions
- **Error propagation:** Ensures error messages are correctly typed and filtered

#### Test 4: Client-Initiated Disconnect Cleanup ✅
- Tests `manager.unsubscribe()` behavior
- Validates subscription stops processing messages after unsubscribe
- Confirms no message leakage post-cleanup
- Verifies proper resource deallocation

#### Test 5: Server-Initiated Disconnect Handling ✅
- Simulates server closing connection with code 1011 ("server-shutdown")
- Validates client receives close event
- Tests zero-reconnection configuration (`maxReconnectAttempts: 0`)
- Confirms connection pool cleanup

### Test Infrastructure
- **Test Runner:** Node.js built-in `node:test` module
- **Assertions:** `node:assert/strict` for type-safe validations
- **WebSocket Library:** `ws` (industry-standard, already in dependencies)
- **Mock Server:** `WebSocketServer` on dynamically allocated port (avoids conflicts)
- **TypeScript:** Full strict mode compliance, no implicit `any` types

### Key Design Patterns
1. **beforeEach/afterEach:** Fresh server instance per test (isolated state)
2. **Dynamic Port Allocation:** Eliminates port conflicts (`port: 0`)
3. **Promise-based Helpers:** `waitForConnection()`, `waitForServerToClose()`, `waitForMessageCount()`
4. **Connection Tracking:** `activeConnections` Set for cleanup verification
5. **Timeout Handling:** Graceful timeout resolution (no test hangs)

### Dependencies
- ✅ Uses existing `ws` library (no new dependencies)
- ✅ Imports from compiled `.genie/mcp/dist/websocket-manager.js`
- ✅ Compatible with existing test patterns (matches `oauth2-auth.test.ts` structure)

---

### Run Tests
```bash
cd forge/mcp
npx tsx --test tests/integration/websocket-streaming.test.ts
```
---

### CODE REVIEW

File: forge/mcp/tests/integration/websocket-streaming.test.ts [hunk 1]
```diff
@@ -0,0 +1,353 @@
+import assert from 'node:assert/strict';
+import type { AddressInfo } from 'net';
+import { beforeEach, afterEach, describe, test } from 'node:test';
+import { WebSocketServer, WebSocket } from 'ws';
+import { setTimeout as delay } from 'node:timers/promises';
+import { WebSocketManager } from '../../../../.genie/mcp/dist/websocket-manager.js';
```
[the remark this answers is not in the document]
Comment (namastex888) [reply forge/mcp/tests/integration/websocket-streaming.test.ts]: @dhairyajangir please verify this feedback, can you run localyl and test?
